apm-lambda-extension/main.go

```diff
@@ -192,7 +191,6 @@ func main() {
 			close(funcDone)
 			backgroundDataSendWg.Wait()
 			if config.SendStrategy == extension.SyncFlush {
-				// Flush APM data now that the function invocation has completed
 				extension.FlushAPMData(client, agentDataChannel, config)
 			}
```

Comment:
Probably out of scope for this PR, but can we just move this to the `case <-funcDone:` case of the background data sending go routine?
The benefit is that the APM Server client is then only executed from a single thread and we don't have to add locks or other mechanisms to ensure visibility.

We might even want to move the background goroutine to `apm_server.go`.

In `main.go`, we would then just be calling `backgroundDataSendWg := extension.startBackgroundSending(config.SendStrategy)`.

`main.go` would just enqueue events to the `agentData` channel and `apm_server.go` is responsible for consuming from that channel. This feels like a better separation of producer/consumer concerns.